### DISCUSSION TITLE
Venetian Blinds Help

### DISCUSSION THREAD

**Discussion by crllsn** [arendst/Tasmota#12873] [Q&A] [answered]
Hi to everyone.
I'd like to setup some sonoff dual R3 to control my venetian blinds.
They work like this:
Venetian closed with closed slats 
if you push UP button for the first 2 seconds the slats open, then it goes up for 60s.
then if you push DOWN for the first 2 seconds it closes the slats and the goes down for 60s.
So it goes up with slats open and down with slats closed.
If I use cursors everything goes ok working as expected (I defined two shutters as per instructions), but if I use real buttons on the wall it cannot know slats tilt.
How can I have slats tilt updated when using wall buttons?
Any idea?
Many tx if someone can help!
Alessandro

EDIT: I think I have to detach buttons and make a rule to control shutter2 for the first 2 seconds, then shutter1. Any hint on how to do this?

**Comment by TrDA-hab** [comment]
Give us a WWW link to the manufacturer's page with your Venetian blinds. We cannot help you without this knowledge.

**Reply by crllsn** [reply]
On the website you won't see anything useful.
What info do you need?
The blinds act as described.
Start condition: Blind fully open with fully open slats
If you push DOWN button, slats fully close (2s) and then start to move down (while moving down slats are fully closed)
At any position you can stop and open the slats pushing UP button.
When it's fully closed, if you push UP button, slats fully open (2s) and then start to move up (while moving up slats are fully open).
So one single motor for both movements. 
Now I have configured it correctly (from the web interface everything is working ok) but if you use wall buttons, tasmota cannot know if you are using buttons to change tilt or to open/close the blind.
Do you need something more / any other info? Just ask!-)
Many thanks in advance!

**Reply by crllsn** [reply]
I think that something like this should work:
when button UP is pressed 
     set a timer = ShutterOpenDuration2
     button moves shutter2
     when timer expires, button moves shutter1

Same logic for DOWN direction.
I'm quite new to Tasmota, do you think I can do this?

**Reply by TrDA-hab** [reply]
If you do not provide the requested information, we will not be able to help you.

**Reply by crllsn** [reply]
https://www.hella.info/en

But honestly I did not understand what info do you need. Just tell me if you need something specific and I can give it to you.

My model is AR 92 Z

**Reply by barbudor** [reply]
You may also give all your config. Not only GPIOs assignment but also any configuration commands you may have set from default configuration.

**Reply by crllsn** [reply]
22:34:56.586 CMD: Grp 0, Cmnd 'BACKLOG', Idx 1, Len 26, Data 'Template; Module; GPIO 255'
22:34:56.749 SRC: Backlog
22:34:56.752 CMD: Grp 0, Cmnd 'TEMPLATE', Idx 1, Len 0, Data ''
22:34:56.768 MQT: stat/tasmota_4932C4/RESULT = {"NAME":"Sonoff Dual R3","GPIO":[32,0,0,0,0,0,0,0,0,576,225,0,0,0,0,0,0,0,0,0,0,7296,7328,224,0,0,0,0,160,161,0,0,0,0,0,0],"FLAG":0,"BASE":1}
22:34:56.954 SRC: Backlog
22:34:56.957 CMD: Grp 0, Cmnd 'MODULE', Idx 1, Len 0, Data ''
22:34:56.971 MQT: stat/tasmota_4932C4/RESULT = {"Module":{"0":"Sonoff Dual R3"}}
22:34:57.159 SRC: Backlog
22:34:57.161 CMD: Grp 0, Cmnd 'GPIO', Idx 1, Len 3, Data '255'
22:34:57.188 MQT: stat/tasmota_4932C4/RESULT = {"GPIO0":{"32":"Button1"},"GPIO1":{"0":"None"},"GPIO2":{"0":"None"},"GPIO3":{"0":"None"},"GPIO4":{"0":"None"},"GPIO5":{"0":"None"},"GPIO9":{"0":"None"},"GPIO10":{"0":"None"},"GPIO12":{"0":"None"},"GPIO13":{"576":"LedLink_i"},"GPIO14":{"225":"Relay2"},"GPIO15":{"0":"None"},"GPIO16":{"0":"None"},"GPIO17":{"0":"None"},"GPIO18":{"0":"None"},"GPIO19":{"0":"None"},"GPIO20":{"0":"None"},"GPIO21":{"0":"None"},"GPIO22":{"0":"None"},"GPIO23":{"0":"None"},"GPIO24":{"0":"None"},"GPIO25":{"7296":"CSE7761 Tx"},"GPIO26":{"7328":"CSE7761 Rx"},"GPIO27":{"224":"Relay1"},"GPIO28":{"0":"None"},"GPIO29":{"0":"None"},"GPIO30":{"0":"None"},"GPIO31":{"0":"None"},"GPIO32":{"160":"Switch1"},"GPIO33":{"161":"Switch2"},"GPIO34":{"0":"None"},"GPIO35":{"0":"None"},"GPIO36":{"0":"None"},"GPIO37":{"0":"None"},"GPIO38":{"0":"None"},"GPIO39":{"0":"None"}}


Status 0 output:

05:17:46.960 MQT: stat/tasmota_4932C4/STATUS0 = {"Status":{"Module":0,"DeviceName":"Cucina","FriendlyName":["Cucina",""],"Topic":"tasmota_4932C4","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"0","SwitchMode":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0},"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota32/release/tasmota32.bin","RestartReason":"Software reset CPU","Uptime":"1T22:26:10","StartupUTC":"2021-08-14T05:51:36","Sleep":0,"CfgHolder":4617,"BootCount":10,"BCResetTime":"2021-07-16T22:03:57","SaveCount":72},"StatusFWR":{"Version":"9.5.0(tasmota)","BuildDateTime":"2021-06-17T08:29:40","Core":"1_0_6","SDK":"v3.3.5-1-g85c43024c","CpuFrequency":80,"Hardware":"ESP32-D0WD-V3","CR":"384/699"},"StatusLOG":{"SerialLog":2,"WebLog":3,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["Piriweb_domus",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["0000800B","2805C8000500060000695A0A000000000000","40000080","00006000","00000000"]},"StatusMEM":{"ProgramSize":1457,"Free":1856,"Heap":201,"PsrMax":0,"PsrFree":0,"ProgramFlashSize":4096,"FlashSize":4096,"FlashFrequency":40,"FlashMode":2,"Features":["00000809","9FDAC787","00152001","B7F7BFCF","05DA9BC0","60360DC7","00084052","20200000","0000002C"],"Drivers":"1,2,3,4,5,7,8,9,10,11,12,14,17,20,21,24,26,27,29,34,50,52","Sensors":"1,2,3,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,28,31,34,37,39,40,42,43,45,51,52,55,56,58,59,64,66,67,74,85,87"},"StatusNET":{"Hostname":"tasmota_4932C4-4804","IPAddress":"192.168.189.18","Gateway":"192.168.189.1","Subnetmask":"255.255.255.0","DNSServer":"192.168.189.1","Mac":"3C:61:05:49:32:C4","Webserver":2,"WifiConfig":0,"WifiPower":17.0},"StatusMQT":{"MqttHost":"192.168.189.200","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_4932C4","MqttUser":"mqtt_user","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":15},"StatusTIM":{"UTC":"2021-08-16T04:17:46","Local":"2021-08-16T05:17:46","StartDST":"2021-03-28T02:00:00","EndDST":"2021-10-31T03:00:00","Timezone":"+01:00","Sunrise":"05:44","Sunset":"20:03"},"StatusPTH":{"PowerDelta":[0,0,0],"PowerLow":0,"PowerHigh":0,"VoltageLow":0,"VoltageHigh":0,"CurrentLow":0,"CurrentHigh":0},"StatusSNS":{"Time":"2021-08-16T05:17:46","Switch1":"ON","Switch2":"ON","ENERGY":{"TotalStartTime":"2021-07-16T22:03:57","Total":0.011,"Yesterday":0.005,"Today":0.000,"Power":[0,0],"ApparentPower":[0,0],"ReactivePower":[0,0],"Factor":[0.00,0.00],"Frequency":0,"Voltage":0,"Current":[0.000,0.000]},"ESP32":{"Temperature":62.8},"Shutter1":{"Position":1,"Direction":0,"Target":1},"Shutter2":{"Position":100,"Direction":0,"Target":100},"TempUnit":"C"},"StatusSTS":{"Time":"2021-08-16T05:17:46","Uptime":"1T22:26:10","UptimeSec":167170,"Heap":199,"SleepMode":"Dynamic","Sleep":0,"LoadAvg":999,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"Piriweb_domus","BSSId":"46:4E:6D:AF:67:38","Channel":4,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:06"}},"StatusSHT":{"SHT0":{"Relay1":1,"Relay2":2,"Open":640,"Close":640,"50perc":50,"Delay":0,"Opt":"0000","Calib":[300,500,700,900,1000],"Mode":"1"},"SHT1":{"Relay1":1,"Relay2":2,"Open":20,"Close":20,"50perc":50,"Delay":0,"Opt":"0000","Calib":[300,500,700,900,1000],"Mode":"1"}}}

**Comment by crllsn** [comment, marked as answer]
Great, Version 10.1 with venetian support seems to do exactly what I tried to explain!-)
Thanks!!
